addons/hr_timesheet/report/report_timesheet_templates.xml

```diff
@@ -92,4 +94,49 @@
         <field name="binding_model_id" ref="model_account_analytic_line"/>
         <field name="binding_type">report</field>
     </record>
+
+    <!-- Project Task Timesheet Report -->
+    <template id="report_project_task_timesheet">
+        <t t-call="web.html_container">
+            <t t-call="web.external_layout">
+                <t t-set="show_Task" t-value="len(docs.mapped('id')) == 1"/>
+                <div class="page">
+                    <t t-foreach="docs" t-as="doc">
+                        <div class="oe_structure"/>
+                        <div class="row" style="margin-top:10px;">
+                            <div class="col-lg-12">
+                                <t t-if="doc.allow_timesheets and doc.timesheet_ids">
+                                    <h1 class="mt-4 mb-4">
+                                        <t t-if="not show_Task">
+                                            Task: <span t-field="doc.name"/>
+                                        </t>
+                                    </h1>
+                                    <h2>
+                                        <span>Timesheet Entries
+                                            <t t-if="show_Task">
+                                                for the <t t-esc="doc.name"/> Task
+                                            </t>
+                                        </span>
+                                    </h2>
+                                    <t t-set='is_uom_day' t-value='doc.encode_uom_in_days'/>
+                                    <t t-set='docs' t-value='doc.timesheet_ids'/>
```

Comment:
Maybe change the variable name here, as docs is already used, it may occur ambiguity if this code needs further improvement or maintenance.